File: BettingApp/src/main/resources/frontend/angular-app/src/app/sport/models/MatchModel.ts
import { TeamModel } from './TeamModel';
import {OddsModel} from "./OddsModel";
import {CompetitionModel} from "./CompetitionModel";
import {AreaModel} from "./AreaModel";


export class MatchModel {
  id: number;
  matchday: number;
  stage: string;
  utcDate: string;
  homeTeam: TeamModel;
  awayTeam: TeamModel;
  competition: CompetitionModel;
  odds_generated: OddsModel;
  area: AreaModel;
  sportType: string;

  constructor(id: number, matchday: number, stage: string, utcDate: string, homeTeam: TeamModel, awayTeam: TeamModel,
              competition: CompetitionModel, odds_generated: OddsModel, area: AreaModel,
              sportType: string) {
    this.id = id;
    this.matchday = matchday;
    this.stage = stage;
    this.utcDate = utcDate;
    this.homeTeam = homeTeam;
    this.awayTeam = awayTeam;
    this.odds_generated = odds_generated;
    this.competition = competition;
    this.area = area;
    this.sportType = sportType;
  }
}

File: BettingApp/src/main/java/com/betuiasi/server/persistence/model/MatchModel.java
package com.betuiasi.server.persistence.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Column;
import lombok.*;

import java.util.HashMap;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class MatchModel {
    @JsonProperty("area")
    AreaModel area;
    @JsonProperty("competition")
    CompetitionModel competition;
    @JsonProperty("season")
    SeasonModel season;
    @JsonProperty("id")
    Long id;
    @JsonProperty("sportType")
    String sportType;
    @JsonProperty("utcDate")
    String utcDate;
    @JsonProperty("status")
    String status;
    @JsonProperty("minute")
    String minute;
    @JsonProperty("injuryTime")
    String injuryTime;
    @JsonIgnore
    @JsonProperty("attendance")
    Object attendance;
    @JsonProperty("matchday")
    int matchday;
    @JsonProperty("stage")
    String stage;
    @JsonProperty("group")
    String group;
    @JsonProperty("lastUpdated")
    String lastUpdated;
    @JsonProperty("homeTeam")
    TeamModel homeTeam;
    @JsonProperty("awayTeam")
    TeamModel awayTeam;
    @JsonProperty("score")
    ScoreModel score;
    @JsonProperty("goals")
    List<GoalModel> goals;
    @JsonIgnore
    @JsonProperty("penalties")
    Object penalties;
    @JsonIgnore
    @JsonProperty("substitutions")
    Object substitutions;
    @JsonIgnore
    @JsonProperty("odds")
    Object odds;
    @JsonIgnore
    @JsonProperty("referees")
    Object referees;
    @JsonProperty("odds_generated")
    OddsModel odds_generated;

}
